qiita_db/metadata_template.py

```diff
@@ -1590,6 +1590,25 @@ def create(cls, md_template, raw_data, study, data_type,
             conn_handler.execute(
                 "DELETE FROM qiita.prep_template where "
                 "{0} = %s".format(cls._id_column), (prep_id,))
+
```

Comment:
Not that I can think of if we want to do it the way that was discussed, which is that we only do the checks if insertion fails.